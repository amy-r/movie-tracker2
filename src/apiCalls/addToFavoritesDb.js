export const addToFavoritesDb = async (movieData, userId) => {
  // eslint-disable-next-line
  const favoriteData = { ...movieData, user_id: userId };
  const url = "api/users/favorites/new";
  try {
    const response = await fetch(url, {
      method: "POST",
      body: JSON.stringify(favoriteData),
      headers: { "Content-Type": "application/json" }
    });
    const favoriteId = await response.json();
    return favoriteId;
  } catch (error) {
    throw new Error("Error adding to favorites");
  }
};
