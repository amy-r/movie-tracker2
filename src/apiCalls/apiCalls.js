import apiKey from './apiKey';
import { movieCleaner } from './movieCleaner'

export const url = `https://api.themoviedb.org/3/discover/movie?api_key=${apiKey}&primary_release_date.gte=2018-02-27&primary_release_date.lte=2018-03-27`;

export const getMovies = async (url) => {
  try {
    const response = await fetch(url);
    const movies = await response.json();
    const cleanMovies = movieCleaner(movies.results);
    return cleanMovies;
  } catch (error) {
    throw new Error('Error getting movies');
  }
};

export const userLogin = async credentials => {
  const url = 'api/users';
  try {
    const response = await fetch(url, {
      method: 'POST',
      body: JSON.stringify(credentials),
      headers: { 'Content-Type': 'application/json' }
    });
    const user = await response.json();
    return user.data;
  } catch (error) {
    throw error;
  }
};

export const userSignup = async accountInfo => {
  const url = 'api/users/new';
  try {
    const response = await fetch(url, {
      method: 'POST',
      body: JSON.stringify(accountInfo),
      headers: { 'Content-Type': 'application/json' }
    });
    const userId = await response.json();
    return userId;
  } catch (error) {
    throw error;
  }
};
