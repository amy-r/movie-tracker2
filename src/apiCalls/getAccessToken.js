import { getTokenData } from "./getTokenData";

export const getAccessToken = async () => {
  const accessToken = sessionStorage.getItem('accessToken');
  const decodedToken = accessToken ? getTokenData(accessToken) : null;
  const expirationTime = decodedToken ? decodedToken.exp : null;
  const now = Math.trunc(Date.now() / 1000);

  // Token is expired or doesn't exist
  if (expirationTime && now >= expirationTime) {
    const apiRoot = process.env.REACT_APP_AUTH_API_URL;
    const url = apiRoot + "/auth/refresh";
    try {
      const response = await fetch(url, {
        method: "GET",
        headers: { "Content-Type": "application/json" },
        credentials: "include"
      });
      const tokens = await response.json();
      sessionStorage.setItem("accessToken", tokens.accessToken);
      return tokens.accessToken;
    } catch (error) {
      throw new Error("Error getting access token");
    }
  // Token is not expired
  } else if (now < expirationTime) {
    return accessToken;
  }
};