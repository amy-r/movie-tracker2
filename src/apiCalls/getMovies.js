const apiKey = process.env.REACT_APP_MDB_API_KEY

export const getMovies = async () => {
  const urlRoot = "https://api.themoviedb.org/3/movie/now_playing?";
  const urlKey = `api_key=${apiKey}`;
  const urlQuery = "&language=en-US&page=1";
  const url = `${urlRoot}${urlKey}${urlQuery}`;

  try {
    const response = await fetch(url);
    const movies = await response.json();
    return movies.results;
  } catch (error) {
    throw new Error("Error getting movies");
  }
};
