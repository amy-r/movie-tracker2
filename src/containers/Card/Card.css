.card {
  box-sizing: border-box;
  margin: 15px 15px;
  width: 220px;
  height: 350px;
  background-color: #1481BA;
}

.card img {
  position: relative;
  width: 200px;
  height: 300px;
  top: 10px;
  filter: drop-shadow(5px 5px 5px)

}

.card button {
  position: relative;
  margin-right: -30px;
  bottom: 340px;
  right: -56px;
}

.favorite img {
  /* border: 2px solid black; */
  /* filter: grayscale(100%) drop-shadow(10px 10px 10px); */
  width: 202px;
  height: 302px;
  box-shadow: 0px 0px 20px 5px #EDBE5F;
}

.favorite .movieDetails {
  top: -311px;
  left: 9px;
  width: 202px;
  height: 302px;
}

.addFaveError, .movieDetails {
  position: relative;
  top: -310px;
  left: 10px;
  width: 200px;
  height: 300px;
  background-color: rgba(255, 255, 255, 0.8);
}

.addFaveError {
  top: -663px;
  padding-top: 50px;
  height: 250px;
}

.addFaveError a {
  color: blue;
  background: none;
}

.movieDetails {
  opacity: 0;
}

.movieDetails:hover {
  opacity: 1;
}

.hidden {
  visibility: hidden;
}

.title, .overview {
  position: relative;
  top: 15px;
  padding: 0 15px;
  margin-bottom: 15px;
}

.overview {
  top: 0;
  height: 240px;
  overflow: auto;
}

.title h4 {
  margin: 15px 0;
  color: #001021;
}

.movieDetails p {
  font-size: 14px;
}

.rating {
  position: absolute;
  left: -5px;
  bottom: -30px;
  width: 200px;
  text-align: left;
}

img.addFavoriteIcon {
  height: 50px;
  width: 50px;
  top: -330px;
  right: -65px;
  z-index: 96;
  cursor: pointer;
  box-shadow: none;
}

.favorite img.addFavoriteIcon {
  filter: brightness(150%);
}

img.addFavoriteIcon:hover {
  height: 50px;
  width: 55px;
  top: -335px;
}

img.addFavoriteIcon:hover .addFaveError {
  width: 202px;
  top: -680px;
}