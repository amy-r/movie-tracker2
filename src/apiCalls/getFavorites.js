/* eslint-disable camelcase */
import { getAccessToken } from "./getAccessToken";

export const getFavorites = async userId => {
  const token = await getAccessToken();
  if (!token) {
    return null;
  }
  const apiRoot = process.env.REACT_APP_AUTH_API_URL;
  const url = apiRoot + `/users/${userId}/favorites`;
  try {
    const response = await fetch(url, {
      method: "GET",
      headers: { 
        "Content-Type": "application/json",
        "Authorization": `Bearer ${token}`
      }
    });
    const favorites = await response.json();
    const cleanFavorites = favorites.data.favorites.map(favorite => {
      return {
        id: favorite.id,
        user_id: favorite.user_id,
        movie_id: parseInt(favorite.movie_id, 10),
        created_at: favorite.created_at
      };
    });
    return cleanFavorites;
  } catch (error) {
    throw new Error("Error getting favorites");
  }
};
