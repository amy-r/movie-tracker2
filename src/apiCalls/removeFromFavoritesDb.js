/* eslint-disable camelcase */
import { getAccessToken } from "./getAccessToken";

export const removeFromFavoritesDb = async (movieId, userId) => {
  const token = await getAccessToken();
  if (!token) {
    return null;
  }
  const favoriteData = { movie_id: movieId, user_id: userId };
  const apiRoot = process.env.REACT_APP_AUTH_API_URL;
  const url = apiRoot + "/users/favorites";
  try {
    const response = await fetch(url, {
      method: "DELETE",
      body: JSON.stringify(favoriteData),
      headers: { 
        "Content-Type": "application/json",
        "Authorization": `Bearer ${token}`
      }
    });
    return await response.json();
  } catch (error) {
    throw new Error("Error removing favorite");
  }
};
