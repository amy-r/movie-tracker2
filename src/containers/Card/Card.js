/* eslint-disable camelcase */
import './Card.css';
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from "react-redux";
import { setError, 
  addFavoriteToStore, 
  removeFavoriteFromStore } from '../../actions';
import { removeFromFavoritesDb } from '../../apiCalls/removeFromFavoritesDb';
import { addToFavoritesDb } from '../../apiCalls/addToFavoritesDb';
import PropTypes from 'prop-types';
import favSelected from '../../assets/fav-selected.png';

export class Card extends Component {
  constructor(props) {
    super(props);

    this.state = {
      favErrorHidden: true
    };
  }

  favoriteError() {
    if (!this.props.user.id) {
      this.setState({ favErrorHidden: false });
    }
  }

  addFavorite() {
    if (this.props.user.id && !this.props.isFavorite) {
      addToFavoritesDb(this.props.movie, this.props.user.id);
      this.props.addFavoriteToStore(this.props.movie);
    } 
  }

  removeFavorite() {
    if (this.props.user.id && this.props.isFavorite) {
      removeFromFavoritesDb(this.props.movie.movie_id, this.props.user.id);
      this.props.removeFavoriteFromStore(this.props.movie.movie_id);
    }
  }

  handleClick = () => {
    this.favoriteError();
    this.addFavorite();
    this.removeFavorite(); 
  };

  render() {
    const { title, overview, poster_path, vote_average } = this.props.movie;
    const favorite = this.props.isFavorite ? 'favorite' : '';
    const { favErrorHidden } = this.state;

    return (
      <div className={`card ${favorite}`}>
        <img src={`https://image.tmdb.org/t/p/w200/${poster_path}`} 
          alt="movie poster" />
        <div className="movieDetails">
          <div className="title">
            <h4>{title}</h4>
          </div>
          <div className="overview">
            <p>{overview}</p>
          </div>
          <div className="rating">
            <h4>Rating: {vote_average}</h4>
          </div>
        </div>
        <img className="addFavoriteIcon" 
          onClick={this.handleClick} 
          src={favSelected} 
          alt="add to favorites button" />
        <div className={"addFaveError"} hidden={favErrorHidden}>
          <p>You must</p>
          <h4><Link to="/login">Login</Link></h4>
          <p>or</p>
          <h4><Link to="/signup">Sign Up</Link></h4>
          <p>before adding favorites.</p>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  user: state.user
});

export const mapDispatchToProps = dispatch => ({
  setError: error => dispatch(setError(error)),
  addFavoriteToStore: favorite => dispatch(addFavoriteToStore(favorite)),
  removeFavoriteFromStore: movie_id => 
    dispatch(removeFavoriteFromStore(movie_id))
});

Card.propTypes = {
  movie: PropTypes.shape({
    title: PropTypes.string,
    overview: PropTypes.string,
    poster_path: PropTypes.string,
    vote_average: PropTypes.num,
    movie_id: PropTypes.num
  }),
  user: PropTypes.object,
  setError: PropTypes.func,
  addFavoriteToStore: PropTypes.func,
  isFavorite: PropTypes.bool,
  removeFavoriteFromStore: PropTypes.func,
  movie_id: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(Card);
