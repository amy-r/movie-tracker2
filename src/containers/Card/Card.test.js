import React from 'react';
import { Card, mapStateToProps, mapDispatchToProps } from "./Card";
import { shallow } from 'enzyme';
import { 
  mockMovieData,  
  mockUser } from '../../mockData/mockData';
import { addToFavoritesDb } from '../../apiCalls/addToFavoritesDb';
import { removeFromFavoritesDb } from '../../apiCalls/removeFromFavoritesDb';

jest.mock('../../apiCalls/addToFavoritesDb.js');
jest.mock('../../apiCalls/removeFromFavoritesDb.js');

describe('Card', () => {
  let wrapper;
  let setError;
  let mockAddFavoriteToStore;
  let mockRemoveFavoriteFromStore;

  beforeEach(() => {
    setError = jest.fn();
    mockAddFavoriteToStore = jest.fn();
    mockRemoveFavoriteFromStore = jest.fn();
    
    wrapper = shallow(
      <Card
        movie={mockMovieData}
        user={mockUser}
        setError={setError}
        addFavoriteToStore={mockAddFavoriteToStore}
        removeFavoriteFromStore={mockRemoveFavoriteFromStore}
        isFavorite={false} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state', () => {
    expect(wrapper.state('favErrorHidden')).toEqual(true);
  });

  it('should update state on favoriteError when there is no user ', () => {
    const mockEmptyUser = {};
    wrapper = shallow(<Card movie={mockMovieData} user={mockEmptyUser} />);
    wrapper.instance().favoriteError();
    expect(wrapper.state('favErrorHidden')).toEqual(false); 
  });

  it('calls addFavoritesToDb and ToStore on addFavorite w/params', () => {
    wrapper.instance().addFavorite();
    expect(addToFavoritesDb).toHaveBeenCalledWith(mockMovieData, mockUser.id);
    expect(mockAddFavoriteToStore).toHaveBeenCalledWith(mockMovieData);
  });

  it('calls removeFavesFromDb and FromStore on removeFaves w/params', () => {
    wrapper = shallow(<Card movie={mockMovieData} 
      user={mockUser} 
      isFavorite={true} 
      removeFavoriteFromStore={mockRemoveFavoriteFromStore}/>);
    wrapper.instance().removeFavorite();
    expect(removeFromFavoritesDb).toHaveBeenCalledWith(mockMovieData.movie_id,
      mockUser.id);
    expect(mockRemoveFavoriteFromStore).toHaveBeenCalledWith(mockMovieData.movie_id); 
  });

  it('calls favoriteError on handleClick', () => {
    const spyFavoriteError = jest.spyOn(wrapper.instance(), 'favoriteError');
    wrapper.instance().handleClick();
    expect(spyFavoriteError).toHaveBeenCalled();
  });

  it('calls addFavorite on handleClick', () => {
    const spyAddFavorite = jest.spyOn(wrapper.instance(), 'addFavorite');
    wrapper.instance().handleClick();
    expect(spyAddFavorite).toHaveBeenCalled();
  });

  it('calls removeFavorite on handleClick', () => {
    const spyRemoveFavorite = jest.spyOn(wrapper.instance(), "removeFavorite");
    wrapper.instance().handleClick();
    expect(spyRemoveFavorite).toHaveBeenCalled();
  });

  it('should map to the store', () => {
    const mockStore = { user: mockUser };
    const mapped = mapStateToProps(mockStore);
    expect(mapped).toEqual(mockStore);
  });

  it('should call dispatch function when using mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.setError();
    mapped.addFavoriteToStore();
    mapped.removeFavoriteFromStore();
    expect(mockDispatch).toHaveBeenCalled();
  });
});
