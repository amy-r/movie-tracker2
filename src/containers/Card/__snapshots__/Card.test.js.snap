// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    addFavoriteToStore={[Function]}
    isFavorite={false}
    movie={
        Object {
            "isFavorite": false,
            "movie_id": 338970,
            "overview": "Lara Croft, the fiercely independent daughter of",
            "poster_path": "/ePyN2nX9t8SOl70eRW47Q29zUFO.jpg",
            "release_date": "2018-03-08",
            "title": "Tomb Raider",
            "vote_average": 6.2,
          }
    }
    removeFavoriteFromStore={[Function]}
    setError={[Function]}
    user={
        Object {
            "email": "tman2272@aol.com",
            "id": 1,
            "name": "Taylor",
            "password": "password",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="movie poster"
          src="https://image.tmdb.org/t/p/w200//ePyN2nX9t8SOl70eRW47Q29zUFO.jpg"
/>,
        <div
          className="movieDetails"
>
          <div
                    className="title"
          >
                    <h4>
                              Tomb Raider
                    </h4>
          </div>
          <div
                    className="overview"
          >
                    <p>
                              Lara Croft, the fiercely independent daughter of
                    </p>
          </div>
          <div
                    className="rating"
          >
                    <h4>
                              Rating: 
                              6.2
                    </h4>
          </div>
</div>,
        <img
          alt="add to favorites button"
          className="addFavoriteIcon"
          onClick={[Function]}
          src="fav-selected.png"
/>,
        <div
          className="addFaveError"
          hidden={true}
>
          <p>
                    You must
          </p>
          <h4>
                    <Link
                              replace={false}
                              to="/login"
                    >
                              Login
                    </Link>
          </h4>
          <p>
                    or
          </p>
          <h4>
                    <Link
                              replace={false}
                              to="/signup"
                    >
                              Sign Up
                    </Link>
          </h4>
          <p>
                    before adding favorites.
          </p>
</div>,
      ],
      "className": "card ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "movie poster",
          "src": "https://image.tmdb.org/t/p/w200//ePyN2nX9t8SOl70eRW47Q29zUFO.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="title"
>
              <h4>
                            Tomb Raider
              </h4>
</div>,
            <div
              className="overview"
>
              <p>
                            Lara Croft, the fiercely independent daughter of
              </p>
</div>,
            <div
              className="rating"
>
              <h4>
                            Rating: 
                            6.2
              </h4>
</div>,
          ],
          "className": "movieDetails",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h4>
                Tomb Raider
</h4>,
              "className": "title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tomb Raider",
              },
              "ref": null,
              "rendered": "Tomb Raider",
              "type": "h4",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                Lara Croft, the fiercely independent daughter of
</p>,
              "className": "overview",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lara Croft, the fiercely independent daughter of",
              },
              "ref": null,
              "rendered": "Lara Croft, the fiercely independent daughter of",
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h4>
                Rating: 
                6.2
</h4>,
              "className": "rating",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Rating: ",
                  6.2,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Rating: ",
                6.2,
              ],
              "type": "h4",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "add to favorites button",
          "className": "addFavoriteIcon",
          "onClick": [Function],
          "src": "fav-selected.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              You must
</p>,
            <h4>
              <Link
                            replace={false}
                            to="/login"
              >
                            Login
              </Link>
</h4>,
            <p>
              or
</p>,
            <h4>
              <Link
                            replace={false}
                            to="/signup"
              >
                            Sign Up
              </Link>
</h4>,
            <p>
              before adding favorites.
</p>,
          ],
          "className": "addFaveError",
          "hidden": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "You must",
            },
            "ref": null,
            "rendered": "You must",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/login"
>
                Login
</Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Login",
                "replace": false,
                "to": "/login",
              },
              "ref": null,
              "rendered": "Login",
              "type": [Function],
            },
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "or",
            },
            "ref": null,
            "rendered": "or",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/signup"
>
                Sign Up
</Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Sign Up",
                "replace": false,
                "to": "/signup",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": [Function],
            },
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "before adding favorites.",
            },
            "ref": null,
            "rendered": "before adding favorites.",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="movie poster"
            src="https://image.tmdb.org/t/p/w200//ePyN2nX9t8SOl70eRW47Q29zUFO.jpg"
/>,
          <div
            className="movieDetails"
>
            <div
                        className="title"
            >
                        <h4>
                                    Tomb Raider
                        </h4>
            </div>
            <div
                        className="overview"
            >
                        <p>
                                    Lara Croft, the fiercely independent daughter of
                        </p>
            </div>
            <div
                        className="rating"
            >
                        <h4>
                                    Rating: 
                                    6.2
                        </h4>
            </div>
</div>,
          <img
            alt="add to favorites button"
            className="addFavoriteIcon"
            onClick={[Function]}
            src="fav-selected.png"
/>,
          <div
            className="addFaveError"
            hidden={true}
>
            <p>
                        You must
            </p>
            <h4>
                        <Link
                                    replace={false}
                                    to="/login"
                        >
                                    Login
                        </Link>
            </h4>
            <p>
                        or
            </p>
            <h4>
                        <Link
                                    replace={false}
                                    to="/signup"
                        >
                                    Sign Up
                        </Link>
            </h4>
            <p>
                        before adding favorites.
            </p>
</div>,
        ],
        "className": "card ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "movie poster",
            "src": "https://image.tmdb.org/t/p/w200//ePyN2nX9t8SOl70eRW47Q29zUFO.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
>
                <h4>
                                Tomb Raider
                </h4>
</div>,
              <div
                className="overview"
>
                <p>
                                Lara Croft, the fiercely independent daughter of
                </p>
</div>,
              <div
                className="rating"
>
                <h4>
                                Rating: 
                                6.2
                </h4>
</div>,
            ],
            "className": "movieDetails",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h4>
                  Tomb Raider
</h4>,
                "className": "title",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Tomb Raider",
                },
                "ref": null,
                "rendered": "Tomb Raider",
                "type": "h4",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  Lara Croft, the fiercely independent daughter of
</p>,
                "className": "overview",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Lara Croft, the fiercely independent daughter of",
                },
                "ref": null,
                "rendered": "Lara Croft, the fiercely independent daughter of",
                "type": "p",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h4>
                  Rating: 
                  6.2
</h4>,
                "className": "rating",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Rating: ",
                    6.2,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Rating: ",
                  6.2,
                ],
                "type": "h4",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "add to favorites button",
            "className": "addFavoriteIcon",
            "onClick": [Function],
            "src": "fav-selected.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                You must
</p>,
              <h4>
                <Link
                                replace={false}
                                to="/login"
                >
                                Login
                </Link>
</h4>,
              <p>
                or
</p>,
              <h4>
                <Link
                                replace={false}
                                to="/signup"
                >
                                Sign Up
                </Link>
</h4>,
              <p>
                before adding favorites.
</p>,
            ],
            "className": "addFaveError",
            "hidden": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "You must",
              },
              "ref": null,
              "rendered": "You must",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/login"
>
                  Login
</Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Login",
                  "replace": false,
                  "to": "/login",
                },
                "ref": null,
                "rendered": "Login",
                "type": [Function],
              },
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "or",
              },
              "ref": null,
              "rendered": "or",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/signup"
>
                  Sign Up
</Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Sign Up",
                  "replace": false,
                  "to": "/signup",
                },
                "ref": null,
                "rendered": "Sign Up",
                "type": [Function],
              },
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "before adding favorites.",
              },
              "ref": null,
              "rendered": "before adding favorites.",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
